agent/lib_aws_sdk_php.c

```diff
@@ -14,10 +14,411 @@
+    'WaitTimeSeconds' => integer,
+));
+
+$result = $client->sendMessage(array(
+    // QueueUrl is required
+    'QueueUrl' => 'string',
+    // MessageBody is required
+    'MessageBody' => 'string',
+    'DelaySeconds' => integer,
+    'MessageAttributes' => array(
+        // Associative array of custom 'String' key names
+        'String' => array(
+            'StringValue' => 'string',
+            'BinaryValue' => 'string',
+            'StringListValues' => array('string', ... ),
+            'BinaryListValues' => array('string', ... ),
+            // DataType is required
+            'DataType' => 'string',
+        ),
+        // ... repeated
+    ),
+));
+
+
+$result = $client->sendMessageBatch(array(
+    // QueueUrl is required
+    'QueueUrl' => 'string',
+    // Entries is required
+    'Entries' => array(
+        array(
+            // Id is required
+            'Id' => 'string',
+            // MessageBody is required
+            'MessageBody' => 'string',
+            'DelaySeconds' => integer,
+            'MessageAttributes' => array(
+                // Associative array of custom 'String' key names
+                'String' => array(
+                    'StringValue' => 'string',
+                    'BinaryValue' => 'string',
+                    'StringListValues' => array('string', ... ),
+                    'BinaryListValues' => array('string', ... ),
+                    // DataType is required
+                    'DataType' => 'string',
+                ),
+                // ... repeated
+            ),
+        ),
+        // ... repeated
+    ),
+));
+
+//clang-format on
+*/
+
+char* nr_lib_aws_sdk_php_get_command_name(NR_EXECUTE_PROTO) {
+  zval* command_name = NULL;
+  char* command_name_string = NULL;
+
+  /* get the command_name. */
+  command_name = nr_php_arg_get(1, NR_EXECUTE_ORIG_ARGS);
+
+  if (0 != nr_php_is_zval_non_empty_string(command_name)) {
+    command_name_string = nr_strdup(Z_STRVAL_P(command_name));
+  }
+
+  nr_php_arg_release(&command_name);
+  return command_name_string;
+}
+
+void nr_lib_aws_sdk_php_sqs_handle(nr_segment_t* segment,
+                                   char* command_name_string,
+                                   NR_EXECUTE_PROTO) {
+  char* command_arg_value = NULL;
+  bool instrumented = false;
+  nr_segment_message_params_t message_params = {
+      .library = "SQS",
+      .destination_type = NR_MESSAGE_DESTINATION_TYPE_QUEUE,
+      .messaging_system = "aws_sqs",
+  };
+  nr_segment_cloud_attrs_t cloud_attrs = {0};
+
+  if (NULL == segment) {
+    return;
+  }
+
+  /* Determine if we instrument this command. */
+  if (nr_streq(command_name_string, "sendMessage")
+      || nr_streq(command_name_string, "sendMessageBatch")) {
+    message_params.message_action = NR_SPAN_PRODUCER;
+    instrumented = true;
+  } else if (nr_streq(command_name_string, "receiveMessage")) {
+    message_params.message_action = NR_SPAN_CONSUMER;
+    instrumented = true;
+  }
```

Comment:
Would it make sense to catch the `else` case and flag it or return if we hit it?

Reply:
Nice idea.
`instrumented` bool and functionality were part of an earlier logic, so removing these as unneeded artifacts from that refactor.
Please see:
[e27135a](https://github.com/newrelic/newrelic-php-agent/pull/1003/commits/e27135a7898690d106fdfb09c3a6c06ffefb439a)